index.js

```diff
@@ -0,0 +1,75 @@
+const express = require('express');
+const server = express()
+const cors = require('cors')
+const db = require('./database/dbHelpers.js')
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+const middleWare = require('./middleware.js')
+server.use(express.json());
+server.use(cors());
+server.use(session({
+  name: 'notsession', // default is connect.sid
+  secret: 'nobody tosses a dwarf!',
+  cookie: {
+    maxAge: 1 * 24 * 60 * 60 * 1000
+  }, // 1 day in milliseconds
+  httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+  resave: false,
+  saveUninitialized: false,
+}));
+
+
+// basic get request to test if server is up 
+
+server.get('/', (req, res) => {
+  res.send("It's alive");
+});
+
+//this route should be protected, only authenticated users should be able to see it
+server.get('/api/users',middleWare.protect, (req, res) => {
+  db.getUsers()
+    .then(users => {
+      res.json(users);
+    })
+    .catch(err => res.send(err))});
+
+server.post('/api/register', (req, res) => {
+  const user = req.body;
+  user.password = bcrypt.hashSync(user.password, 16) // 16 denotes how many times its hashes
+  db.insertUser(user)
+    .then(id => {
+      res.status(201).json({ id: id[0] })
+    })
+    .catch(err => {
+      res.status(500).json({ error: 'Please Try Again' })
+    });
+});
+
+server.post('/api/login', (req, res) => {
+  const creds = req.body;
```

Comment:
I'd consider tweaking this variable name to read as `credentials` or `user` for clarity 👍